service_crategen/src/commands/generate/codegen/query.rs

```diff
@@ -240,6 +240,18 @@ fn value_name(service: &Service, shape: &Shape) -> String {
 }
 
 fn member_location(service: &Service, member: &Member, default: &str) -> String {
+    // Seems a bit hacky to avoid doing this just for EC2:
+    match service.shape_for_member(member) {
```

Comment:
Nit: Using `if let Some(ref shape) = service.shape_for_member(member)` might be shorter here than `match`, and you could avoid the no-op `None` branch.